framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/resource/impl/ProjectImpl.java

```diff
@@ -98,6 +100,58 @@ public void processTemplateAndCreateResources(URL templateUrl, Map<String, Strin
         }
     }
 
+    @Override
+    public synchronized void processApbRun(String image, Map<String, String> extraVars) {
+        String podName = "apb-pod-" + UUID.randomUUID().toString().substring(0, 4);
+
+        OpenShiftBinaryClient oc = OpenShiftBinaryClient.getInstance();
+        oc.project(projectName);
+        if(util.getServiceAccount("apb") == null) {
+            oc.executeCommand("Creating serviceaccount failed.", "create", "serviceaccount", "apb");
+            oc.executeCommand("Role binding failed.", "create", "rolebinding", "apb", "--clusterrole=admin", "--serviceaccount=" + projectName + ":apb");
+        }
+
+        List<String> args = new ArrayList<>();
+        args.add("run");
+        args.add(podName);
+        // args.add("--namespace=" + projectName);
+        args.add("--env=POD_NAME=" + podName);
+        args.add("--env=POD_NAMESPACE=" + projectName);
+        args.add("--image=" + image);
+        args.add("--restart=Never");
+        args.add("--attach=true");
+        args.add("--serviceaccount=apb");
+        args.add("--");
+        args.add("provision");
+        args.add("--extra-vars");
+        args.add(formatExtraVars(extraVars));
+
+        logger.info("Executing command: oc " + getApbCommand(args));
+        oc.executeCommand("APB failed.", args.toArray(new String[args.size()]));
+    }
+
+    private String getApbCommand(List<String> args) {
+        String out = "";
+        out = args.stream().map((arg) -> arg + " ").reduce(out, String::concat);
+        return out;
+    }
+
+    private String formatExtraVars(Map<String, String> extraVars) {
+        StringBuilder extraVarsArg = new StringBuilder("{");
+        extraVars.forEach((String key, String value) -> {
```

Comment:
What about replacing this with something like:
`extraVars.entrySet().stream().map(entry -> "\"" + entry.getKey() + "\":\"" + entry.getValue() + "\"")
                       .collect(Collectors.joining(", ", "{", "}"));`